src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -10287,85 +10287,85 @@ namespace SurfaceGeometry {
                 ErrorsFound = true;
             }
 
+            enum Month
             {
-                auto const SELECT_CASE_var(state.dataSurface->StormWindow(StormWinNum).MonthOn);
-
-                if ((SELECT_CASE_var == 1) || (SELECT_CASE_var == 3) || (SELECT_CASE_var == 5) || (SELECT_CASE_var == 7) || (SELECT_CASE_var == 8) ||
-                    (SELECT_CASE_var == 10) || (SELECT_CASE_var == 12)) {
-                    if (state.dataSurface->StormWindow(StormWinNum).DayOfMonthOn > 31) {
-                        ShowSevereError(state,
-                                        format("{}: Date On (Day of Month) [{}], invalid for WindowProperty:StormWindow Input #{}",
-                                               cCurrentModuleObject,
-                                               state.dataSurface->StormWindow(StormWinNum).DayOfMonthOn,
-                                               StormWinNum));
-                        ErrorsFound = true;
-                    }
-                } else if ((SELECT_CASE_var == 4) || (SELECT_CASE_var == 6) || (SELECT_CASE_var == 9) || (SELECT_CASE_var == 11)) {
-                    if (state.dataSurface->StormWindow(StormWinNum).DayOfMonthOn > 30) {
-                        ShowSevereError(state,
-                                        format("{}: Date On (Day of Month) [{}], invalid for WindowProperty:StormWindow Input #{}",
-                                               cCurrentModuleObject,
-                                               state.dataSurface->StormWindow(StormWinNum).DayOfMonthOn,
-                                               StormWinNum));
-                        ErrorsFound = true;
-                    }
-                } else if (SELECT_CASE_var == 2) {
-                    if (state.dataSurface->StormWindow(StormWinNum).DayOfMonthOn > 29) {
-                        ShowSevereError(state,
-                                        format("{}: Date On (Day of Month) [{}], invalid for WindowProperty:StormWindow Input #{}",
-                                               cCurrentModuleObject,
-                                               state.dataSurface->StormWindow(StormWinNum).DayOfMonthOn,
-                                               StormWinNum));
-                        ErrorsFound = true;
-                    }
-                } else {
+                January = 1,
+                February,
+                March,
+                April,
+                May,
+                June,
+                July,
+                August,
+                September,
+                October,
+                November,
+                December
+            };
+            constexpr std::array<int, 13> oneBasedDaysInMonth = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+            switch (state.dataSurface->StormWindow(StormWinNum).MonthOn) {
```

Comment:
I hate to bring this up but ... `BITF` would be faster here. Actually, you could also just do a range check.

Reply:
I'd prefer to avoid the BITF macro if possible, but would be happy to do a range check.

Reply:
That's good too. At some point,  The `BITF` macros should be switched to `constexpr` functions or function templates. That will also make their APIs cleaner since you would be able to do `bitfTestAny(value, enum1, enum2, enum3, enum4)` rather than `BITF_TEST_ANY(BITF(value), BITF(enum1) | BITF(enum2) | BITF(enum3) | BITF(enum4))`.  But not now.

Reply:
It's not used very frequently yet. It can wait.